Makefile

```diff
@@ -1021,6 +1021,11 @@ endif
 TEST_PACKAGES ?= ./...
 
 test:
+	# ensure Go's build cache for the cleanercmd is fresh so that tests don't have to build from scratch. This
+	# could take some time and counts against the test's timeout, which can lead to flakes.
+	# c.f. https://github.com/coder/internal/issues/1026
+	mkdir -p build
+	$(GIT_FLAGS) go build -o ./build/cleaner github.com/coder/coder/v2/coderd/database/dbtestutil/cleanercmd
```

Comment:
Could we make this a make target instead?

Reply:
The point here is to prime Go's build cache, not to actually create the binary artifact. (Is there a way to tell go to discard the artifact? `-o /dev/null`?) `make` isn't going to be aware of the dependencies of the cleaner without a bunch of fragile work; since `go` handles that under the covers.

I could make it a phony target if you thought it would help readability...